pysages/methods/umbrella_integration.py

```diff
@@ -165,12 +165,11 @@ def submit_work(executor, method, context_args, callback):
         )
 
     futures = []
-    with executor as ex:
-        for rep, submethod in enumerate(method.submethods):
-            local_context_args = deepcopy(context_args)
-            local_context_args["replica_num"] = rep
-            callback = method.histograms[rep]
-            futures.append(submit_work(ex, submethod, local_context_args, callback))
+    for rep, submethod in enumerate(method.submethods):
```

Comment:
Why is this removing the context manager?

Reply:
Because the context manager calls `shutdown` when you are leaving.
And an MPI pool cannot be restarted after `shutdown` was called.

Which is fine for Umbrella integration, where we use the pool once.
But for the string you want to run multiple umbrella integrations and that requires using the MPI pool multiple times.

Reply:
Got it. I'd still want to have this probably run conditionally, but we can figure out a way to do so later.

Reply:
I also checked and calling `result()` on a future ensures that it is done with its work.
So we shouldn't have sync issues.

Reply:
Yeah, maybe we add something that optionally shutdowns the communicator at the end of a run.
It's probably better to explicitly release the resources.

Reply:
@pabloferz are you OK with this?

Reply:
Yes, this is fine for now. We can find a way to be explicit about calling `shutdown` later.

Reply:
It is actually quite simple to implement the shutdown.
So I did this here: https://github.com/SSAGESLabs/PySAGES/pull/182/commits/c2dfb817ee43a096785daec1342a4a9bc508f186
Pending tests, I think that should do it for us.

Reply:
Let me know what you think.

Reply:
That works for me!